metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -493,10 +521,110 @@ def emit_chart_mces(self) -> Iterable[MetadataWorkUnit]:
                     title=chart_data.get("slice_name", ""),
                     entity_urn=chart_snapshot.urn,
                 )
+            except Exception as e:
+                logger.warning(
+                    f"Failed to emit dataset MCE {chart_data.get('table_name')}. Error: \n{e}"
+                )
+                continue
+
+    def gen_schema_fields(self, column_data: List[Dict[str, str]]) -> List[SchemaField]:
+        schema_fields: List[SchemaField] = []
+        for col in column_data:
+            col_type = (col.get("type") or "").lower()
+            data_type = SUPERSET_FIELD_TYPE_MAPPINGS.get(col_type)
+            field = SchemaField(
+                fieldPath=col.get("column_name", ""),
+                type=SchemaFieldDataType(data_type() if data_type else NullType()),
+                nativeDataType="",
+                description=col.get("column_name", ""),
+                nullable=True,
+            )
+            schema_fields.append(field)
+        return schema_fields
+
+    def gen_schema_metadata(
+        self,
+        dataset_response: dict,
+    ) -> SchemaMetadata:
+        dataset_response = dataset_response.get("result", {})
+        column_data = dataset_response.get("columns", [])
+        schema_metadata = SchemaMetadata(
+            schemaName=dataset_response.get("table_name", ""),
+            platform=make_data_platform_urn(self.platform),
+            version=0,
+            hash="",
+            platformSchema=MySqlDDL(tableSchema=""),
+            fields=self.gen_schema_fields(column_data),
+        )
+        return schema_metadata
+
+    def gen_dataset_urn(self, datahub_dataset_name: str) -> str:
+        return make_dataset_urn_with_platform_instance(
+            platform=self.platform,
+            name=datahub_dataset_name,
+            platform_instance=self.config.platform_instance,
+            env=self.config.env,
+        )
+
+    def construct_dataset_from_dataset_data(
+        self, dataset_data: dict
+    ) -> DatasetSnapshot:
+        dataset_response = self.get_dataset_info(dataset_data.get("id"))
+        dataset = SupersetDataset(**dataset_response["result"])
+        datasource_urn = self.get_datasource_urn_from_id(
+            dataset_response, self.platform
+        )
+        if dataset.changed_on_utc:
+            modified_dt = dp.parse(dataset.changed_on_utc)
+        else:
+            modified_dt = datetime.now(timezone.utc)
```

Comment:
this type of logic should also get moved into the SupersetDataset

Reply:
looks like this one got missed - we can use a pydantic validator to automatically parse the timestamp

Also, I don't think defaulting to datetime.now() makes sense - can we just leave it as None / missing if we don't have info?

Reply:
Agreed, this was logic I copied over from dashboard/charts. I'll look to clean that up in the future when we refactor those. I've worked in logic to handle None type